src/dataflow/src/logging/timely.rs

```diff
@@ -345,8 +376,18 @@ pub fn construct<A: Allocate>(
             }
         });
 
-        let channels = channels.as_collection();
         let addresses = addresses.as_collection();
+        let channels = channels.as_collection().map(
+            |(id, worker, source_node, source_port, target_node, target_port)| {
+                (
+                    (id, worker),
```

Comment:
I think you could just `session.give` the data with this structure to save on a map operator? not a huge deal

Reply:
Huh, I think you're right. Will try!